sim/common/itemhelpers/weaponprocs.go

```diff
@@ -55,89 +55,82 @@ func CreateWeaponProcDamage(itemId int32, itemName string, ppm float64, spellId
 		}
 
 		procSpell := character.RegisterSpell(sc)
-		procMask := character.GetProcMaskForItem(itemId)
-		ppmm := character.AutoAttacks.NewPPMManager(ppm, procMask)
-
-		character.GetOrRegisterAura(core.Aura{
-			Label:    fmt.Sprintf("%s Proc Aura", itemName),
-			Duration: core.NeverExpires,
-			OnReset: func(aura *core.Aura, sim *core.Simulation) {
-				aura.Activate(sim)
-			},
-			OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
-				if spell.Flags.Matches(spellFlagExclude) { // This check is needed because it doesn't use aura_helpers ApplyProcTriggerCallback
-					return
-				}
-				if result.Landed() && ppmm.Proc(sim, spell.ProcMask, aura.Label) {
-					procSpell.Cast(sim, result.Target)
-				}
+
+		aura := core.MakeProcTriggerAura(&character.Unit, core.ProcTrigger{
+			Name:              fmt.Sprintf("%s Proc Aura", itemName),
+			Callback:          core.CallbackOnSpellHitDealt,
+			Outcome:           core.OutcomeLanded,
+			DPM:               character.AutoAttacks.NewDynamicProcManagerForWeaponEffect(itemID, ppm, 0),
+			DPMProcType:       core.DPMProcNoWeaponSpecials,
```

Comment:
Wait I thought weapons did proc from specials?

Reply:
I've renamed this to make it a bit more clear.
Currently the default for `ProcTriggerAura` is `dpm.ProcWithWeaponSpecials()`. 
However not all procs that could be simplified to a `ProcTriggerAura` use the `dpm.Proc()`, this property should be used to indicate this nuance.